File: Java/Tester.java
import java.io.*;
import java.util.Random;

public class Tester {

  public static void main(String[] args) {

    for ( int i = 0; i < 10000; i++ ) {
      float a = random_float();
      if ( !check_float( a ) ) {
        System.out.println( "失敗" );
        return; 
      } 
    }
    System.out.println( "成功" );
  }

  private static float random_float() {
    Random r = new Random(); 
    int k = r.nextInt(50);
    double d = r.nextDouble();  
    if ( r.nextBoolean() ) {
      return (float)( k * d );
    } else {
      return - (float)( k * d );
    }
  }

  private static boolean check_float( float a ) {

    int a_int = Fpu.getUint32_t( a );
    return Fpu.is_valid_fsqrt( a_int ) && Fpu.is_valid_finv( a_int ); 
  }

}

File: C/tester.c
#include "fpu.h"
#include <math.h>
#include <time.h>

int check_float( float a ) {
  uint32_t a_int = getUint32_t( a );
  return is_valid_fsqrt( a_int ) && is_valid_finv( a_int );
}

int rand_float() {
  float f = (float)rand() / (float)rand();
  if ( (int)rand() % 2 == 0 ) {
    f *= -1; 
  } 
  return f;
}

int main(int argc, char const* argv[])
{
  srand((unsigned)time(NULL));
  int i;
  for ( int i = 0; i < 100000; i++ ) {
    if ( !check_float( rand_float() ) ) {
      printf( "失敗\n" );
      return 0; 
    } 
  }
  printf( "成功\n" );
  return 0;
   
}
